test/framework/src/main/java/org/elasticsearch/test/rest/Stash.java

```diff
@@ -81,7 +84,26 @@ public boolean isStashedValue(Object key) {
      * as arguments for following requests (e.g. scroll_id)
      */
     public Object getValue(String key) throws IOException {
-        Object stashedValue = stashObjectPath.evaluate(key.substring(1));
+        if (key.charAt(0) == '$' && key.charAt(1) != '{') {
+            return unstash(key.substring(1));
+        }
+        Matcher m = EXTENDED_KEY.matcher(key);
+        /*
+         * String*Buffer* because that is what the Matcher API takes. In modern versions of java the uncontended synchronization is very,
+         * very cheap so that should be a problem. The 16 is just a guess, but it is the same guess that StringBuffer makes when you
+         * initialize it with a string like you are going to append to it.
+         */
+        StringBuffer b = new StringBuffer(key.length() + 16);
```

Comment:
but what id we drop the 16 or the argument at all and remove this long comment? do we need this optimization here?